### PR TITLE
Provide functionality for validating quality files. 

### LINKED ISSUE #688: Build in functionality for validating quality edit test files.
The validate quality edit test files method in each file generator class will adjust quality edit test files so that they pass syntax and validity edits. The goal of this issue would aim to build that functionality into the generate error files scripts.

### PR DESCRIPTION
**Pull request by MichaelFurey17**
Close #688

### CODE REVIEW

File: 2018/python/file_generator.py [hunk 1]
```diff
@@ -351,7 +352,7 @@ def validate_quality_edit_file(self, quality_filename):
 			res_df = pd.DataFrame(checker.results)
 			new_df = res_df[(res_df['status']=='failed')]
 
-			#If there are no syntax or validity edits,
+			#If there are no syntax or validity edits or row_ids is not in column names,
```
Comment (Kibrael) [inline 2018/python/file_generator.py]: Can you rephrase this so it is more accessible?

File: 2019/python/file_generator.py [hunk 2]
```diff
@@ -351,7 +352,7 @@ def validate_quality_edit_file(self, quality_filename):
 			res_df = pd.DataFrame(checker.results)
 			new_df = res_df[(res_df['status']=='failed')]
 
-			#If there are no syntax or validity edits,
+			#If there are no syntax or validity edits or row_ids is not in column names,
```
Comment (Kibrael) [inline 2019/python/file_generator.py]: Same request for clarity as above.

File: 2018/python/file_generator.py (hunk 1 above)
Comment (MichaelFurey17) [reply 2018/python/file_generator.py]: Sure! Some edits that are not row specific generate an edit report without row_ids as a column. These edits were not being generated correctly, and so I decided to exclude them as checks in the validation function here. 

However, I think I need to resolve the issue with this PR here.

File: 2018/python/file_generator.py [hunk 3]
```diff
@@ -356,11 +360,15 @@ def validate_quality_edit_file(self, quality_filename):
 			res_df = pd.DataFrame(checker.results)
 			new_df = res_df[(res_df['status']=='failed')]
 
-			#If there are no syntax or validity edits,
-			#the data is written to a new directory for quality 
-			#test files that pass syntax and validity edits. 
+			# The function would ignore TS edits. The edit
```
Comment (Kibrael) [inline 2018/python/file_generator.py]: I am confused by these comments, do you mean that the TS row is dropped? Is there a certain condition that causes that to happen?

File: 2018/python/rules_engine.py [hunk 4]
```diff
@@ -72,7 +72,7 @@ def results_wrapper(self, fail_df=None, field_name=None, edit_name=None, row_typ
 				failed_rows = list(fail_df.uli)
 				self.update_results(edit_name=edit_name, edit_field_results=result, row_type=row_type, fields=field_name, row_ids=failed_rows, fail_count=count)
 			else:
-				self.update_results(edit_name=edit_name, edit_field_results=result, row_type=row_type, fields=field_name)
+				self.update_results(edit_name=edit_name, edit_field_results=result, row_type=row_type, fields=field_name, row_ids='TS', fail_count=1)
```
Comment (Kibrael) [inline 2018/python/rules_engine.py]: Can you leave a comment on why you're hardcoding the fail count to 1?

File: 2019/python/file_generator.py [hunk 5]
```diff
@@ -356,11 +360,15 @@ def validate_quality_edit_file(self, quality_filename):
 			res_df = pd.DataFrame(checker.results)
 			new_df = res_df[(res_df['status']=='failed')]
 
-			#If there are no syntax or validity edits,
-			#the data is written to a new directory for quality 
-			#test files that pass syntax and validity edits. 
+			# The function would ignore TS edits. The edit
```
Comment (Kibrael) [inline 2019/python/file_generator.py]: Do you mean: this function ignores TS edits and drops results related to edit fails from the TS?

File: 2019/python/rules_engine.py [hunk 6]
```diff
@@ -72,7 +72,7 @@ def results_wrapper(self, fail_df=None, field_name=None, edit_name=None, row_typ
 				failed_rows = list(fail_df.uli)
 				self.update_results(edit_name=edit_name, edit_field_results=result, row_type=row_type, fields=field_name, row_ids=failed_rows, fail_count=count)
 			else:
-				self.update_results(edit_name=edit_name, edit_field_results=result, row_type=row_type, fields=field_name)
+				self.update_results(edit_name=edit_name, edit_field_results=result, row_type=row_type, fields=field_name, row_ids='TS', fail_count=1)
```
Comment (Kibrael) [inline 2019/python/rules_engine.py]: Can you add a comment explaining why the TS fail count was hardcoded to 1?

File: 2019/python/generate_2019_error_files.py [hunk 7]
```diff
@@ -1,9 +1,27 @@
-#This script will generate a file that will fail a specific syntax, validity, or quality edit (though others may also fail)
+import os
```
Comment (Kibrael) [inline 2019/python/generate_2019_error_files.py]: Good job working out how to make the quality edit test files pass validity edits.

File: 2019/python/file_generator.py (hunk 5 above)
Comment (MichaelFurey17) [reply 2019/python/file_generator.py]: Yes, that would be the function below. I can update the comments to your wording.

File: 2019/python/generate_2019_error_files.py (hunk 7 above)
Comment (MichaelFurey17) [reply 2019/python/generate_2019_error_files.py]: Thanks!

File: 2018/python/file_generator.py (hunk 3 above)
Comment (MichaelFurey17) [reply 2018/python/file_generator.py]: The function ignores TS edits - I can explain more fully.

File: 2019/python/rules_engine.py (hunk 6 above)
Comment (MichaelFurey17) [reply 2019/python/rules_engine.py]: Sure!

File: 2018/python/rules_engine.py (hunk 4 above)
Comment (MichaelFurey17) [reply 2018/python/rules_engine.py]: Sure!
